corehq/apps/userreports/views.py

```diff
@@ -1478,11 +1485,29 @@ def export_data_source(request, domain, config_id):
     config, _ = get_datasource_config_or_404(config_id, domain)
     adapter = get_indicator_adapter(config, load_source='export_data_source')
     url = reverse('export_configurable_data_source', args=[domain, config._id])
-    return export_sql_adapter_view(request, domain, adapter, url)
+    return export_sql_adapter_view(request, adapter, url)
 
 
-def export_sql_adapter_view(request, domain, adapter, too_large_redirect_url):
-    q = adapter.get_query_object()
+def _get_db_query_from_user_params(query, params):
+    """Constructs a query from the provided user parameters
+
+    When pagination parameters are specified (both `offset` and `limit`), the query will be sorted in ascenting
+    order using `inserted_at`. If only one pagination parameter is specified, it will not have any effect on the
+    query. Both `offset` and `limit` must be specified for it to take effect.
+    """
+
+    query = query.filter_by(**params.keyword_filters)
+    for sql_filter in params.sql_filters:
+        query = query.filter(sql_filter)
+
+    if params.offset and params.limit:
```

Comment:
If we specify an offset or limit of 0, then this would resolve to `False`. Also, what would happen if we specified negative numbers, for example a limit of -5?

Reply:
> If we specify an offset or limit of 0, then this would resolve to False

The query params are being read as strings, which is why this will not evaluate to `False`. However, I think I should still update the code to make the intent clear so that future readers wouldn't get confused

> Also, what would happen if we specified negative numbers, for example a limit of -5

This (same with the offset parameter)
![image](https://github.com/dimagi/commcare-hq/assets/64970009/6f72874c-0445-4712-b415-da5d87638137)

Reply:
Lol, I didn't think. Scratch the commit

Reply:
This is the correct one: https://github.com/dimagi/commcare-hq/pull/33340/commits/780e01e143ebb2bc5f7e3843d77a9c1a6394670d

Reply:
Thanks for the clarification @SmittieC!